src/www/app/net.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright 2021 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {FakeOutlineTunnel} from './fake_tunnel';
+import {Tunnel} from './tunnel';
+
+// Exposes native networking capabilities to the web app.
+export interface NativeNetworking {
```

Comment:
The reason for separating them is to limit complexity and increase maintainability.
The way it is now, it's not clear that the implementations for reachability and tunnel are separate. Having them separate makes it clear that there are no interactions and that you can replace them independently.

It's more composable. If I would like to provide a different reachability implementation, now I need to provide a full NativeNetworking. Same if I want to provide a new vpn tunnel. It prevents things like creating a wrapper for the net test that would log the results of the tests, for example.

I think it's great that the `createServer` is now inside outline_server. You don't really need to change much. Instead of injecting one object (NativeNetworking) in the repository, you would be injecting two (ConnectivityTest,  TunnelProvider). They would come from the OutlinePlatform, which would provide two methods `getConnectivityTest` and `getTunnelProvider`.

But I'll leave it up to you. I think this PR is an improvement over the status quo.

Reply:
By the way, a good example is if we wanted to pick whether to use the Go stack or the libev stack.
I could have two different TunnelProvider that I would choose based on a flag. The reachability would be the same. But this design would force us to implement two separate NativeNetworking instead.

Reply:
Done in 6713055.